staging/src/k8s.io/legacy-cloud-providers/azure/azure_managedDiskController.go

```diff
@@ -346,6 +346,11 @@ func (c *Cloud) GetAzureDiskLabels(diskURI string) (map[string]string, error) {
 		return nil, err
 	}
 
+	// no azure credential is set, return nil
+	if c.DisksClient == nil {
```

Comment:
also skip when `c==nil`?

Reply:
if `c==nil`, then `c.GetAzureDiskLabels()` already crashed ...

Reply:
why can DisksClient be nil? Is that even a valid situation for `Cloud`?

Reply:
@sttts that's because azure identity is not set in `azure.json` for `kube-apiserver`, so DisksClient is not initialized, and it does not make sense to initialize DisksClient if azure identity is not set:

https://github.com/kubernetes/kubernetes/blob/b2247906a4a5118455658e458b7f01f2b27c5b2b/staging/src/k8s.io/legacy-cloud-providers/azure/azure.go#L462-L466

Reply:
@sttts, email Bz1847185 . API server fix might be needed too.

Reply:
>that's because azure identity is not set in azure.json for kube-apiserver,...

It'd be useful to share some context/bug ref in the commit message i.e git commit -m"" -m"here"

Reply:
It'd be nice to include a unit test for this

Reply:
shall we go with this PR first? One of my colleagues are adding unit test for the azure_managedDiskController.go
cc @weijiehu

Reply:
my PR for azure_managedDiskController.go unittest is in progress, I will send it tomorrow.

Reply:
@andyzhangx I new unit test PR here #92216